acceptance/lib/puppet/acceptance/common_utils.rb

```diff
@@ -0,0 +1,152 @@
+module Puppet
+  module Acceptance
+    module CronUtils
+      def clean(agent, o={})
+        o = {:user => 'tstuser'}.merge(o)
+        run_cron_on(agent, :remove, o[:user])
+        apply_manifest_on(agent, %[user { '%s': ensure => absent, managehome => false }] % o[:user])
+      end
+
+      def setup(agent, o={})
+        o = {:user => 'tstuser'}.merge(o)
+        apply_manifest_on(agent, %[user { '%s': ensure => present, managehome => false }] % o[:user])
+        apply_manifest_on(agent, %[case $operatingsystem {
+                                     centos, redhat: {$cron = 'cronie'}
+                                     solaris: { $cron = 'core-os' }
+                                     default: {$cron ='cron'} }
+                                     package {'cron': name=> $cron, ensure=>present, }])
+      end
+    end
+
+    module CAUtils
+
+      def initialize_ssl
+        hostname = on(master, 'facter hostname').stdout.strip
+        fqdn = on(master, 'facter fqdn').stdout.strip
+
+        if master.use_service_scripts?
+          step "Ensure puppet is stopped"
+          # Passenger, in particular, must be shutdown for the cert setup steps to work,
+          # but any running puppet master will interfere with webrick starting up and
+          # potentially ignore the puppet.conf changes.
+          on(master, puppet('resource', 'service', master['puppetservice'], "ensure=stopped"))
+        end
+
+        step "Clear SSL on all hosts"
+        hosts.each do |host|
+          ssldir = on(host, puppet('agent --configprint ssldir')).stdout.chomp
+          on(host, "rm -rf '#{ssldir}'")
+        end
+
+        step "Master: Start Puppet Master" do
+          master_opts = {
+            :main => {
+              :dns_alt_names => "puppet,#{hostname},#{fqdn}",
+            },
+            :__service_args__ => {
+              # apache2 service scripts can't restart if we've removed the ssl dir
```

Comment:
And here it looks like we are working around the above comment about how we removed the entire ssldir and not just the contents